nntrainer/models/neuralnet.cpp

```diff
@@ -162,31 +164,71 @@ int NeuralNetwork::initialize() {
  */
 NeuralNetwork::~NeuralNetwork() { model_graph.reset(); }
 
-void NeuralNetwork::setLabels(sharedConstTensors label) {
-  auto fill_label = [&label](auto const &layer_node) {
-    NNTR_THROW_IF(label.size() != layer_node->getNumOutputs(),
-                  std::invalid_argument)
-      << "label size does not match with the layer requirements"
-      << " layer: " << layer_node->getName() << " label size: " << label.size()
-      << " requirements size: " << layer_node->getNumOutputs();
+static void setLabels(const std::vector<Tensor> &data,
+                      const std::vector<Var_Grad *> &label_list) {
 
-    for (unsigned int i = 0; i < layer_node->getNumOutputs(); i++) {
-      layer_node->getOutputGradUnsafe(i) = *label[i];
-    }
-  };
+  NNTR_THROW_IF(!data.empty() && data.size() != label_list.size(),
+                std::invalid_argument)
+    << "label size does not match with the network requirements"
+    << " label size: " << data.size()
+    << " requirements size: " << label_list.size();
 
-  auto clear_label = [](auto const &layer_node) {
-    for (unsigned int i = 0; i < layer_node->getNumOutputs(); i++) {
-      layer_node->getOutputGradUnsafe(i) = Tensor();
-    }
-  };
+  /// feed or clear label
+  for (unsigned int idx = 0; idx < data.size(); idx++) {
+    if (data.empty())
+      label_list[idx]->initializeGradient();
+    else if (data.size() == 1)
+      label_list[idx]->initializeGradient(data[0]);
```

Comment:
maybe this is not needed?

Reply:
This sets all the given labels to all the output nodes that require labels. Can you elaborate on your question?

Reply:
this line seems to equivalent to the

```
label_list[idx]->initializeGradient(data[idx]);
```
so removing this else if statement?

Reply:
aah, I understand the question. It is possible to receive just 1 label or input which must be set to all the labels/inputs. So, this `if` condition takes care of that special condition.

Reply:
if that's the case, could you check if the loop condition should be ?
```
for (unsigned int idx = 0; idx < label.size(); idx++) {
```

Reply:
You are right, its a bug for particular this function. It was resolved later with #1544, but I will update it here as well.

Reply:
Cool, no worries. You might want to delay to reduce work at will :)

Reply:
its minor, so its fine.